app/hooks/useOrders.tsx

```diff
@@ -59,17 +60,47 @@ export function intoOrders(response: AxiosResponse<Entity>): Order[] {
     });
 }
 
+async function checkIfOrderWasMatched(
+  prevOrders: Order[],
+  orders: Order[]
+): Promise<Order[]> {
+  if (!prevOrders) {
+    return orders;
+  }
+
+  try {
+    for (const prevOrder of prevOrders) {
+      const order = orders.find(o => o.id === prevOrder.id);
+
+      if (order && order.state.open.value !== prevOrder.state.open.value) {
+        await mutate('/swaps');
+      }
+    }
+  } catch (e) {
+    console.error(e);
+  }
+
+  return orders;
+}
+
 export default function useOrders() {
   const cnd = useCnd();
+  let prevOrders = null;
 
   const { data: orders } = useSWR(
     '/orders',
-    key => cnd.fetch(key).then(intoOrders),
+    key =>
+      cnd
+        .fetch(key)
+        .then(intoOrders)
+        .then(orders => checkIfOrderWasMatched(prevOrders, orders)),
```

Comment:
Doing computation in here bit us in the past because useSwr doesn't bubble up errors here on purpose.

I'd rather do all of this after the `useSwr` call.